DataDog/libdatadog
# Add Profiled endpoints statistics

# What does this PR do?

This PR allows to send the profiled endpoints count in the `event.json` file.

The counts should be sent in the `event.json` file:
```
{
 "start":"<START DATE>", 
 "end":"<END DATE>",
 "attachments":["<ATTACHMENT1>", "<ATTACHMENT2>"],
 "tags_profiler":"<PROFILER TAGS>",
 "endpoint_counts": {
   "/api/endpoint": 10,
   "/api/different-endpoint": 35
 },
 "family":"<FAMILY>",
 "version":"4"
}
```

# Motivation

The goal is to be able to see how much resources are used by an average individual call to an endpoint. From those statistics, the backend will do the computation.


# How to test the change?

For .NET, I compiled the code and generated the NuGet to be used by the profiler. I sniffed the HTTP request and checked that:
- Adding endpoints, we can see the `event.json` containing the endpoints stats.
- Not adding endpoints, we can see an empty list.
